refactor(helpers)!: remove default value from arrayElement (#2045)
faker-js/faker: docs/guide/upgrading.md

The `faker.location.zipCodeByState` method has been deprecated, but will also now throw an error if the current locale does not have a `postcode_by_state` definition.

### Methods will throw on empty data set inputs

The methods `faker.helper.arrayElement` and `faker.helper.arrayElements` previously defaulted the `array` argument to a simply string array if non was provided.
This behavior is no longer supported.

Additionally, by providing an empty array argument (`[]`) the functions previously returned `undefined`.
This behavior was undesired and has been removed.
The methods will now throw an `FakerError` instead.

The same thing happens now if you provide an empty object `{}` to `faker.helpers.objectKey` or `faker.helpers.objectValue`.

> 

Reviewer: i think we should have an example of how to fix it if you were reliant on the old behavior.

Something like:
Old
```js
let post = {}
post.tags = faker.helpers.arrayElements(["dogs","cats","fish","horses","sheep"],{min:0, max:3})
post.featuredTag = faker.helpers.arrayElement(post.tags)
```

New
```js
let post = {}
post.tags = faker.helpers.arrayElements(["dogs","cats","fish","horses","sheep"],{min:0, max:3})
post.featuredTag = post.tags.length===0 ? undefined : faker.helpers.arrayElement(post.tags)
```
or
```js
let post = {}
post.tags = faker.helpers.arrayElements(["dogs","cats","fish","horses","sheep"],{min:0, max:3})
try {
  post.featuredTag = faker.helpers.arrayElement(post.tags)
} catch (e) {
  //error thrown, leave post.featuredTag as undefined
}
```

Reviewer 2: Or we define an overload with an option: { allowEmpty: true } that adds `undefined` to the result type.

The `faker.location.zipCodeByState` method has been deprecated, but will also now throw an error if the current locale does not have a `postcode_by_state` definition.

### Methods will throw on empty data set inputs

> The methods `faker.helper.arrayElement` and `faker.helper.arrayElements` previously defaulted the `array` argument to a simply string array if non was provided.

Reviewer: simply to simple, non to none

Reviewer 3: ```suggestion
The methods `faker.helpers.arrayElement` and `faker.helpers.arrayElements` previously defaulted the `array` argument to a simple string array if none was provided.
```

> The methods `faker.helper.arrayElement` and `faker.helper.arrayElements` previously defaulted the `array` argument to a simply string array if non was provided.

Reviewer: helper to helpers

Reviewer 3: ```suggestion
The methods `faker.helpers.arrayElement` and `faker.helpers.arrayElements` previously defaulted the `array` argument to a simple string array if none was provided.
```

This behavior is no longer supported.

Additionally, by providing an empty array argument (`[]`) the functions previously returned `undefined`.
This behavior was undesired and has been removed.
The methods will now throw an `FakerError` instead.

The same thing happens now if you provide an empty object `{}` to `faker.helpers.objectKey` or `faker.helpers.objectValue`.

**Old**

```ts
const allTags = ['dogs', 'cats', 'fish', 'horses', 'sheep'];
const tags = faker.helpers.arrayElements(allTags, { min: 0, max: 3 });
// `tags` might be an empty array which was no problem in v7
const featuredTag = faker.helpers.arrayElement(tags);

> // `featureTag` will will be typed as `string` but could actually be `undefined`

Reviewer: Double "will"

Reviewer 3: ```suggestion
// `featureTag` will be typed as `string` but could actually be `undefined`
```

The `faker.location.zipCodeByState` method has been deprecated, but will also now throw an error if the current locale does not have a `postcode_by_state` definition.

### Methods will throw on empty data set inputs

The methods `faker.helpers.arrayElement` and `faker.helpers.arrayElements` previously defaulted the `array` argument to a simple string array if none was provided.
This behavior is no longer supported, as the default value has been removed.
You are now required to provide an argument.

> Additionally, by providing an empty array argument (`[]`) the functions previously returned `undefined`.

Reviewer 4: ```suggestion
Additionally, when passing in an empty array argument (`[]`), the functions previously returned `undefined`.
```

> This behavior was undesired and has been removed.

Reviewer 2: Maybe mention that this violated the typescript definition regarding the return type.

Reviewer 2: ```suggestion
This behavior violated the expected return type of the method.
```
